src/createAnimatedComponent.tsx

```diff
@@ -409,11 +472,64 @@ export default function createAnimatedComponent(
       }
     }
 
+    _attachInlineStyles() {
+      const newInlineStyles: StyleProps = getInlineStylesFromProps(this.props);
+      const hasChanged = inlineStylesHasChanged(
+        newInlineStyles,
+        this._inlineStyles
+      );
+
+      if (hasChanged) {
+        const sharableViewDescriptors =
+          this._inlineStylesViewDescriptors.sharableViewDescriptors;
+
+        const maybeViewRef = NativeReanimatedModule.native
+          ? undefined
+          : ({ items: new Set([this]) } as ViewRefSet<any>); // see makeViewsRefSet
+
+        const updaterFunction = () => {
+          'worklet';
+          const update: StyleProps = {};
+          for (const [key, styleValue] of Object.entries(newInlineStyles)) {
+            if (key === 'transform') {
+              update[key] = styleValue.map((t: Record<string, any>) =>
+                Object.keys(t).reduce(
+                  (acc, curr) => ({ ...acc, [curr]: t[curr].value }),
+                  {}
```

Comment:
I'd prefer to rewrite this bit using simple loops. One aspect is that object spread isn't typically very fast and another one is readability. Using `map` method is ok, but `reduce` with spread is a bit too much I believe